File: backend/app/schemas/document.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: str
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    status: str
    chunk_count: Optional[int]
    error_message: Optional[str]
    project_id: Optional[str]
    uploaded_by: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentUploadResponse(BaseModel):
    document: DocumentResponse
    message: str


class DocumentSearchRequest(BaseModel):
    query: str
    top_k: int = 5
    project_id: Optional[str] = None


class DocumentChunkResult(BaseModel):
    document_id: str
    document_name: str
    content: str
    score: float
    chunk_index: int

File: frontend/src/types/document.ts
export type DocumentStatus = 'pending' | 'processing' | 'processed' | 'failed'

export interface Document {
  id: string
  filename: string
  original_filename: string
  file_type: string
  file_size: number
  status: DocumentStatus
  chunk_count: number | null
  error_message: string | null
  project_id: string | null
  uploaded_by: string
  created_at: string
  updated_at: string
}

export interface DocumentSearchRequest {
  query: string
  top_k?: number
  project_id?: string
}

export interface DocumentChunkResult {
  document_id: string
  document_name: string
  content: string
  score: number
  chunk_index: number
}
